### PR TITLE
Add initial structure

### PR DESCRIPTION
**Pull request by RDPerera**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: ballerina/Ballerina.toml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+graalvmCompatible = true
+
+[[platform.java17.dependency]]
+groupId = "io.ballerina.lib.sap"
+artifactId = "sap.jco-native"
+version = "0.1.0-SNAPSHOT"
+path = "../native/build/libs/sap.jco-native-0.1.0-SNAPSHOT.jar"
+
+[[platform.java17.dependency]]
+path = "../native/libs/sapidoc3.jar"
```
Comment (niveathika) [inline ballerina/Ballerina.toml:25]: Add provided scope. Or else these libs will be packed in the bala. That will be a license violations of SAP libs

File: ballerina/errors.bal [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        // input is a SAP_JCO 'error' value
+        return errorOrMessage;
+    } else if errorOrMessage is error {
+        // input is a Ballerina 'error' value
+        return error Error(errorOrMessage.message(), detail = errorOrMessage.detail(), cause = errorOrMessage.cause());
+    } else {
+        // input is the error message
+        return error Error(errorOrMessage);
+    }
+}
```
Comment (niveathika) [inline ballerina/errors.bal:31]: Where is this function used? Why cant we do that at the native level?

File: ballerina/init.bal [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import ballerina/log;
+
+function init() {
+    error? module = setModule();
+    if module is error {
+        log:printError("Error in module initialization", module);
+    }
+}
+
+function setModule() returns error? = @java:Method {
```
Comment (niveathika) [inline ballerina/init.bal:27]: Cant we make it isolated? Also are we expecting an error scenario here?

File: examples/build.gradle [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+def graalvmFlag = ""
+
+task clean {
+}
+
+task test {
+    if (project.hasProperty("balGraalVMTest")) {
+        graalvmFlag = "--graalvm"
+    }
```
Comment (niveathika) [inline examples/build.gradle:29]: No need

Comment (niveathika) [reply examples/build.gradle:29]: See updted latest build script https://github.com/ballerina-platform/module-ballerinax-sap.s4hana.sales/blob/main/examples/build.gradle#L21

File: native/src/main/java/module-info.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+ module io.ballerina.lib.sap {
+     requires sapjco3;
+     requires sapidoc3;
+    requires io.ballerina.runtime;
+     requires org.slf4j;
+     requires java.logging;
```
Comment (niveathika) [inline native/src/main/java/module-info.java:24]: Why do we need both slf4j and java.logging?

File: .github/CODEOWNERS [hunk 6]
```diff
@@ -0,0 +1,7 @@
+# Lines starting with '#' are comments.
+# Each line is a file pattern followed by one or more owners.
+
+# See: https://help.github.com/articles/about-codeowners/
+
+# These owners will be the default owners for everything in the repo.
+*       @niveathika @RDPerera
```
Comment (niveathika) [inline .github/CODEOWNERS:7]:
```suggestion
*       @RDPerera @niveathika 
```

File: ballerina/Ballerina.toml [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+keywords = ["Business Management/ERP", "Cost/Paid", "Vendor/SAP"]
+repository = "https://github.com/ballerina-platform/module-ballerinax-sap.jco"
+icon = "icon.png"
+license = ["Apache-2.0"]
+
+[build-options]
+observabilityIncluded = true
+
+[platform.java17]
+graalvmCompatible = true
```
Comment (niveathika) [inline ballerina/Ballerina.toml:16]: Were these properties tested? For now lets remove them and commit

File: ballerina/errors.bal (hunk 2 above)
Comment (RDPerera) [reply ballerina/errors.bal]: Yes, removed with [3b98cc3](https://github.com/ballerina-platform/module-ballerinax-sap.jco/pull/1/commits/3b98cc39afaeb4f6ad60a496009006ee83f96d96)

**Review by niveathika — COMMENTED** [review]
LGTM other than minor comments

File: ballerina/init.bal (hunk 3 above)
Comment (RDPerera) [reply ballerina/init.bal]: Changed with [3b98cc3](https://github.com/ballerina-platform/module-ballerinax-sap.jco/pull/1/commits/3b98cc39afaeb4f6ad60a496009006ee83f96d96).

File: examples/build.gradle (hunk 4 above)
Comment (RDPerera) [reply examples/build.gradle]: Changed with [3b98cc3](https://github.com/ballerina-platform/module-ballerinax-sap.jco/pull/1/commits/3b98cc39afaeb4f6ad60a496009006ee83f96d96).

File: native/src/main/java/module-info.java (hunk 5 above)
Comment (RDPerera) [reply native/src/main/java/module-info.java]: Changed with [3b98cc3](https://github.com/ballerina-platform/module-ballerinax-sap.jco/pull/1/commits/3b98cc39afaeb4f6ad60a496009006ee83f96d96).

File: ballerina/Ballerina.toml (hunk 7 above)
Comment (RDPerera) [reply ballerina/Ballerina.toml]: Changed with [9983b0b](https://github.com/ballerina-platform/module-ballerinax-sap.jco/pull/1/commits/9983b0b7a54c408c1721d199a19a3164283937f4).

**Review by niveathika — APPROVED** [review]

**Pull request merged**